app/helpers/sessions_helper.rb

```diff
@@ -0,0 +1,18 @@
+module SessionsHelper
+  def log_in user
+    session[:user_id] = user.id
+  end
+
+  def current_user
+    @current_user ||= User.find_by id: session[:user_id]
+  end
+
+  def logged_in?
+    !current_user.nil?
```

Comment:
sao phải đảo ngược (!) kết quả vậy?
Em thử tìm hiểu về: nil?, present? và any?
Sau đó thử xác nhận xem có hàm mô phù hợp nhất nhé

Reply:
1. Do hàm logged_in? kiểm tra 1 user đã Login hay chưa (trả về true - tồn tại Object user). Hàm .nil? của Object kiểm tra user nil user(không tồn tại). Nên phải dùng ! để đảo ngược

2. nil? chỉ để kiểm tra 1 Object có nil hay không. any? để kiểm tra có ít nhất 1 user Object được trả về ( NOT nil, NOT false ) . present? để kiểm tra 1 Object NOT nil, NOT false, NOT empty

Ở đây present?  sẽ thích hợp nhất